src/Mvc/RouteDispatcher.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * Quantum PHP Framework
+ *
+ * An open source software development framework for PHP
+ *
+ * @package Quantum
+ * @author Arman Ag. <arman.ag@softberg.org>
+ * @copyright Copyright (c) 2018 Softberg LLC (https://softberg.org)
+ * @link http://quantum.softberg.org/
+ * @since 2.9.5
+ */
+
+namespace Quantum\Mvc;
+
+use Quantum\Exceptions\ControllerException;
+use Quantum\Middleware\MiddlewareExecutor;
+use Quantum\Handlers\ViewCacheHandler;
+use Quantum\Router\RouteController;
+use Quantum\Libraries\Csrf\Csrf;
+use Quantum\Http\Response;
+use Quantum\Loader\Loader;
+use Quantum\Http\Request;
+use Quantum\Di\Di;
+
+class RouteDispatcher
+{
+    /**
+     * Handles the incoming HTTP request and generates a response.
+     *
+     * This includes:
+     * 1. Executing registered middleware.
+     * 2. Attempting to serve a cached view.
+     * 3. Handling routing through either a route callback or a controller action.
+     *
+     * Controller lifecycle hooks `__before` and `__after` will be invoked if defined.
+     * CSRF token verification is performed for applicable HTTP methods.
+     *
+     * @param Request  $request  The incoming HTTP request object.
+     * @param Response $response The HTTP response object to be sent.
+     *
+     * @return void
+     */
+    public static function handle(Request $request, Response $response): void
+    {
+        // 1. Apply middleware
+        [$request, $response] = (new MiddlewareExecutor())->execute($request, $response);
```

Comment:
Looks like this array unpacking of objects is only available from php 7.4 and above, but the framework still supports 7.3 so this will not work and need to be changed to using list().
Unless we drop 7.3 support from framework version 2.9.7 @armanist ?

Reply:
Yes we are still supporting php 7.3 and possibly in next quantum framework version (2.9.7) we still need to support it because once we drop it we may need to revise the complete codebase to correct any outdated code. We might do it from version 3.0 onward.